### PR TITLE
Add idle_timeout to routes API

### PR DESCRIPTION
**Pull request by denismakogon**
Adding `inactivity_timeout` to routes API. 
This change will allow users to define idle timeout for hot functions via API instead of having it hardcoded in `worker.go`.

Closes: #554

### CODE REVIEW

**Comment by treeder** [comment]
Generally looks good, but can we call it `IdleTimeout` instead of `InactivityTimeout` ?

**Comment by denismakogon** [comment]
@treeder Sure, why not.

File: api/models/new_task.go [hunk 1]
```diff
@@ -54,6 +54,11 @@ type NewTask struct {
 
 	*/
 	Timeout *int32 `json:"timeout,omitempty"`
+
+	/* Hot function idle timeout in seconds before termination.
+
+	*/
+	IdleTimeout *int32 `json:"inactivity_timeout,omitempty"`
```
Comment (jmank88) [inline api/models/new_task.go]: There are a few dangling 'inactivity' usages in json fields and errors messages

**Comment by denismakogon** [comment]
@ccirello So, yes, there are two timeouts, and it doesn't look like something bad or confusing. One timeout is for function call, and the idle timeout defines for how long to keep hot function around without serving requests for it. You're saying that it was not thought through, i tend to disagree on this, because it's not clear why hardcoded 30s timeout is exact timeout that every hot function should wait without processing requests.
From user perspective, if i have a workload that comes each 1m, and i don't want to spend time to wait until function will be started (+ overhead of function itself, like pulling out data from 3rd party services, etc.) rather than setting idle timeout for 2m, so with such workload my hot function will be more effective in terms of faster requests serving.
Does it make sense?
cc @treeder @seiflotfy @pedronasser

**Comment by ccirello** [comment]
@denismakogon - makes sense to a point.

> So, yes, there are two timeouts, and it doesn't look like something bad or confusing. 

Without context, I doubt people will know the difference between timeout and idle_timeout. This could be easily addressed with documentation though - as of now, however, docs are missing.

> You're saying that it was not thought through, i tend to disagree on this, because it's not clear why hardcoded 30s timeout is exact timeout that every hot function should wait without processing requests before becoming "cold".

Indeed, it is not clear the reason behind 30s. However, whenever we change a constant to a variable, we are changing the behavior of the system. So what you are saying is that the user needs it - which I am fine with - and that this reason alone is enough to justify this change as it is right now.

I did not see anywhere any consideration of the consequences of this change: it does not define min and max boundaries, does not explain the consequence of too short or too long idle timeouts, it seems to address the zero case implicitly (for instance, it is not self-evident for me what happens if this idle_timeout is zero); also I do not see in the source code the interaction between this field being updated in the DB while the hot functions are still running. 

Again, I am not against the goal of this change - but this implementation of this change is lacking in many aspects. 

A middle-way here would be to merge this as it is, and open a batch of tickets addresses its deficiencies.

**Comment by denismakogon** [comment]
@ccirello I agree on documentation as well as with edge cases. I'll make next PR with docs that explicitly describes all possible cases for idle time and its combinations with regular timeout.

**Comment by treeder** [comment]
The docs should be part of this PR, agree with @ccirello . Not complete without docs.

**Comment by denismakogon** [comment]
@treeder doc is already here. Check last 2 commits.

**Comment by treeder** [comment]
@ccirello ok now with docs?

**Comment by ccirello** [comment]
@denismakogon @treeder LGTM

**Pull request merged**